framework/include/utils/PointReduction.h

```diff
@@ -0,0 +1,29 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "Moose.h"
+#include <vector>
+
+namespace PointReduction
+{
+
+typedef std::pair<Real, Real> FunctionNode;
+typedef std::vector<FunctionNode> FunctionNodeList;
+
+/// compute the perpendicular distance of a point from a line defined by begin and end points
```

Comment:
Document what the parameters and return values of the functions are following the MOOSE conventions.
https://www.mooseframework.org/framework_development/code_standards.html